### ISSUE TITLE
Differentiate between "Project name" and "module name"

### ISSUE THREAD

**Issue by 1ec5** [realm/jazzy#411]
I’m migrating an Objective-C framework from appledoc to jazzy (mapbox/mapbox-gl-native#3203). The framework’s full name is “Mapbox iOS SDK”, while the actual module name that you import is simply `Mapbox`. Based on jazzy’s default templates, the full name seems like the most appropriate value for `module_name`. But if `module_name` contains spaces, the following error occurs:

```
/usr/local/Cellar/ruby/2.2.3/lib/ruby/2.2.0/uri/generic.rb:1100:in `rescue in merge': bad URI(is not URI?): docsets/Mapbox iOS SDK.tgz (URI::InvalidURIError)
    from /usr/local/Cellar/ruby/2.2.3/lib/ruby/2.2.0/uri/generic.rb:1097:in `merge'
    from /usr/local/Cellar/ruby/2.2.3/lib/ruby/2.2.0/uri/rfc3986_parser.rb:88:in `each'
    from /usr/local/Cellar/ruby/2.2.3/lib/ruby/2.2.0/uri/rfc3986_parser.rb:88:in `inject'
    from /usr/local/Cellar/ruby/2.2.3/lib/ruby/2.2.0/uri/rfc3986_parser.rb:88:in `join'
    from /usr/local/Cellar/ruby/2.2.3/lib/ruby/2.2.0/uri/common.rb:264:in `join'
    from /usr/local/lib/ruby/gems/2.2.0/gems/jazzy-0.4.1/lib/jazzy/docset_builder.rb:89:in `block in create_xml'
    from /usr/local/lib/ruby/gems/2.2.0/gems/jazzy-0.4.1/lib/jazzy/docset_builder.rb:88:in `open'
    from /usr/local/lib/ruby/gems/2.2.0/gems/jazzy-0.4.1/lib/jazzy/docset_builder.rb:88:in `open'
    from /usr/local/lib/ruby/gems/2.2.0/gems/jazzy-0.4.1/lib/jazzy/docset_builder.rb:88:in `create_xml'
    from /usr/local/lib/ruby/gems/2.2.0/gems/jazzy-0.4.1/lib/jazzy/docset_builder.rb:33:in `build!'
    from /usr/local/lib/ruby/gems/2.2.0/gems/jazzy-0.4.1/lib/jazzy/doc_builder.rb:130:in `build_docs_for_sourcekitten_output'
    from /usr/local/lib/ruby/gems/2.2.0/gems/jazzy-0.4.1/lib/jazzy/doc_builder.rb:68:in `build'
    from /usr/local/lib/ruby/gems/2.2.0/gems/jazzy-0.4.1/bin/jazzy:15:in `<top (required)>'
    from /usr/local/bin/jazzy:23:in `load'
    from /usr/local/bin/jazzy:23:in `<main>'
make: *** [ipackage] Error 1
```

**Comment by 1ec5** [comment]
#412 fixes this error. However, jazzy should probably have a separate `name` flag for the framework’s full name, independent of the module name.

**Comment by 1ec5** [comment]
The workaround is to specify the actual module name (without spaces) in module_name, then replace every instance of module_name in the templates and optionally move the generated docset via a shell script. But I think it's common enough for a framework to have a different "common name" and module name.

**Comment by nlutsenko** [comment]
+1 Module name and framework name probably should be separate.
Example: `Parse iOS SDK` - full docset name and `Parse` - module name.

**Comment by jpsim** [comment]
Yes, I agree that we should differentiate between "project name" and "module name". I'll update this issue to track that.

**Comment by istx25** [comment]
Any progress on this since Dec 2015?

**Comment by jpsim** [comment]
doesn't look like it

**Comment by johnfairh** [comment]
Fixed in master via #1077
